recording/lib/storage/s3.ex

```diff
@@ -56,4 +60,116 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording.Storage.S3 do
     |> then(&ExAws.Config.new(:s3, &1))
     |> Map.to_list()
   end
+
+  @spec try_to_fix_objects(file_config(), s3_config()) :: :ok | :error
+  def try_to_fix_objects(file_config, s3_config) do
+    with {:ok, objects} <- list_objects(s3_config),
+         :ok <- fix_objects(file_config, s3_config, objects) do
+      :ok
+    else
+      {:error, _reason} = error -> handle_error(error, s3_config.config)
+    end
+  end
+
+  defp fix_objects(file_config, s3_config, objects) do
+    if perform_object_fixes(file_config, s3_config, objects) do
+      :ok
+    else
+      objects = Enum.map(objects, fn {filename, _size} -> filename end)
+
+      case clean_objects(objects, s3_config.opts) do
+        {:ok, _term} -> {:error, :failed_to_fix_objects}
+        {:error, _reason} -> {:error, :failed_to_clean_objects}
+      end
+    end
+  end
```

Comment:
I would move it to `try_to_fix_objects` and handle cleaning in else branch of with.

Reply:
it's not that easy because in else clause you cannot use objects right now